### PR TITLE
[WFLY-15654] Configure use-realm-role-mappings

### PR DESCRIPTION
**Pull request by claudioweiler**
Add the ability to configure use-realm-role-mappings in the elytron-oidc-client subsystem

https://issues.redhat.com/browse/WFLY-15654

### CODE REVIEW

**Review by fjuma — CHANGES_REQUESTED** [review]
@claudioweiler Thanks very much for working on this! This looks great.

I just realized that something that I forgot to mention in WFLY-15654 is that we need to add version 2 of the `elytron-oidc-client` subsystem before making these changes.

I have a more complete explanation of why we need to do that and how to do it here (I've also described how this would impact the location of some of the changes that were made in this PR):
https://issues.redhat.com/browse/WFLY-15710

If that's something you're interested in working on as well, feel free to assign to yourself and that could be included as a separate commit in this PR. I'm also happy to help with any questions about this. Otherwise, I could submit a PR for WFLY-15710 separately and your PR could be updated to depend on it.

File: elytron-oidc-client/subsystem/src/main/java/org/wildfly/extension/elytron/oidc/SecureDeploymentDefinition.java [hunk 1]
```diff
@@ -75,6 +75,12 @@
                     .setAlternatives(ElytronOidcDescriptionConstants.RESOURCE)
                     .build();
 
+    protected static final SimpleAttributeDefinition USE_REALM_ROLE_MAPPINGS =
+            new SimpleAttributeDefinitionBuilder(ElytronOidcDescriptionConstants.USE_REALM_ROLE_MAPPINGS, ModelType.BOOLEAN, true)
+                    .setAllowExpression(true)
+                    .setDefaultValue(ModelNode.TRUE)
+                    .build();
+
```
[the remark this answers is not in the document]
Comment (claudioweiler) [reply elytron-oidc-client/subsystem/src/main/java/org/wildfly/extension/elytron/oidc/SecureDeploymentDefinition.java:83]: Right @fjuma, I see that some required changes will reflect in this current PR, too. I'll try to look at this, in meanwhile I will not push any changes here.

[the remark this answers is not in the document]
Comment (fjuma) [reply elytron-oidc-client/subsystem/src/main/java/org/wildfly/extension/elytron/oidc/SecureDeploymentDefinition.java:83]: Sounds good, thanks @claudioweiler!

**Pull request closed without merging**